code/integration-tests/src/androidTest/java/com/adobe/marketing/mobile/integration/core/RulesEngineIntegrationTests.kt

```diff
@@ -0,0 +1,277 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.integration.core
+
+import android.app.Application
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.adobe.marketing.mobile.AdobeCallbackWithError
+import com.adobe.marketing.mobile.AdobeError
+import com.adobe.marketing.mobile.Event
+import com.adobe.marketing.mobile.LoggingMode
+import com.adobe.marketing.mobile.MobileCore
+import com.adobe.marketing.mobile.SDKHelper
+import com.adobe.marketing.mobile.integration.MockNetworkResponse
+import com.adobe.marketing.mobile.services.Networking
+import com.adobe.marketing.mobile.services.ServiceProvider
+import java.net.HttpURLConnection
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertFalse
+import junit.framework.TestCase.assertTrue
+import org.json.JSONObject
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class RulesEngineIntegrationTests {
+
+    companion object {
+        const val TEST_RULES_URL = "https://rules.com/rules.zip"
+        const val TEST_APP_ID = "appId"
+        const val TEST_CONFIG_URL = "https://assets.adobedtm.com/${TEST_APP_ID}.json"
+        const val TEST_RULES_RESOURCE = "rules_dispatch_consequence.zip"
```

Comment:
I noticed it is missing in Swift side. 
Can you add one test each for attachData, modifyData and dispatch consequence for completion. You can add few more rules and rename `rules_dispatch_consequence.zip -> rules_integration.zip`